.github/workflows/ci.yml

```diff
@@ -38,9 +38,9 @@ jobs:
           # sometimes go cache causes issues with lint
           cache: false
 
-      - uses: golangci/golangci-lint-action@v3
+      - uses: golangci/golangci-lint-action@v6
         with:
-          version: v1.52
+          version: v1.54
```

Comment:
This is the golangci-lint version, right? The latest one seems to be `1.61` can we not use that?

Reply:
I was mostly manually picking the appropriate changes/version from release/0.12 and main branch. We are still on v1.54. Wasn't sure if the newer ones would through newer lint errors or the like.

@helsaawy any preferences on this? Should we use 1.61 or is it ok to keep it at 1.54 similar to main and release/0.12 branches?

Reply:
probably best to stick to 1.54
the issue with the latest linter ones is that they are usually build with the newest go (go1.23), which can cause lint errors/failures if our repo doesn't also use that version

Reply:
@ambarve is it ok if we keep it at v1.54 for now?

Reply:
Yeah, that's okay.